pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,232 @@
+	})
+
+	ginkgo.It("should create nodepool and provision EC2 instances with proper STS permissions", func(ctx context.Context) {
+		var nodes corev1.NodeList
+		expect.NoError(client.List(ctx, &nodes))
+		
+		if len(nodes.Items) == 0 {
+			ginkgo.Skip("No nodes found")
+		}
+
+		clusterName := extractClusterName(nodes.Items)
+		clusterID := getClusterID(nodes.Items)
+		
+		if clusterName == "" || clusterID == "" {
+			ginkgo.Skip("Could not determine cluster information")
+		}
+
+		nodePoolName := fmt.Sprintf("osde2e-sts-test-%d", time.Now().Unix())
+		
+		ginkgo.By("Creating NodePool to test STS permissions for day 2 operations")
+		err := createNodePool(clusterID, nodePoolName)
+		
+		if err != nil {
+			expect.NoError(fmt.Errorf("NodePool creation failed - STS permissions missing %v.", err))
+		}
+
+		// Cleanup
+		defer deleteNodePool(clusterID, nodePoolName)
+
+		ginkgo.By("Waiting for new nodes from NodePool (tests ec2:RunInstances, ec2:CreateTags)")
+		newNodes := waitForNewNodes(ctx, client, len(nodes.Items), nodePoolName)
+		
+		if len(newNodes) == 0 {
+			expect.NoError(fmt.Errorf("NodePool failed to provision nodes - STS permissions (ec2:RunInstances, ec2:CreateTags) missing."))
+		}
+		
+		ginkgo.By("Validating nodes exist in AWS (tests ec2:DescribeInstances)")
+		for _, node := range newNodes {
+			exists, err := aws.CcsAwsSession.CheckIfEC2ExistBasedOnNodeName(node.Name)
+			if err != nil {
+				expect.NoError(fmt.Errorf("AWS validation failed for node %s - STS permissions (ec2:DescribeInstances) missing: %v.", node.Name, err))
+			}
+			if !exists {
+				expect.NoError(fmt.Errorf("Node %s not found in AWS - STS permissions (ec2:DescribeInstances) missing.", node.Name))
+			}
+		}
+
+		ginkgo.By("Testing workload scheduling on new NodePool")
+		err = testWorkload(ctx, client, h, nodePoolName)
+		if err != nil {
+			expect.NoError(fmt.Errorf("Workload scheduling failed on NodePool: %v.", err))
+		}
+		
+		ginkgo.By("NodePool operations completed successfully - STS permissions working")
+		ginkgo.GinkgoLogr.Info("NodePool STS permissions test PASSED", 
+			"cluster", clusterName,
+			"nodepool", nodePoolName,
+			"nodes_created", len(newNodes))
+	})
+})
+
+func extractClusterName(nodes []corev1.Node) string {
+	for _, node := range nodes {
+		if label, exists := node.Labels["hypershift.openshift.io/nodePool"]; exists {
+			if strings.Contains(label, "-workers-") {
+				return strings.Split(label, "-workers-")[0]
+			}
+		}
+	}
+	return ""
+}
+
+func getClusterID(nodes []corev1.Node) string {
+	if clusterID := os.Getenv("CLUSTER_ID"); clusterID != "" {
+		return clusterID
+	}
+	
+	for _, node := range nodes {
+		if strings.Contains(node.Name, "-") {
+			parts := strings.Split(node.Name, "-")
+			if len(parts) > 0 && len(parts[0]) > 10 {
+				return parts[0]
+			}
+		}
+	}
+	return ""
+}
+
+func createNodePool(clusterID, nodePoolName string) error {
+	if _, err := exec.Command("rosa", "whoami").CombinedOutput(); err != nil {
```

Comment:
I don't think we should be shelling out to `rosa` using `exec.Command`.  Aren't there libraries to take advantage of that can do this?